### ISSUE TITLE
Segfault in Vulkan related to DiffuseProbeGrid Visualization 5f7a30f

### ISSUE THREAD

**Issue by RoaldFre** [o3de/o3de#7517]
**Describe the bug**
There seems to be a segfault/null pointer dereference when using Vulkan RHI (currently only tested under Linux), linked to the DiffuseProbeGrid Visualization of commit 5f7a30f8dafac41bc02af4d315661be183e86132

**Steps to reproduce**
1. Checkout commit 5f7a30f8dafac41bc02af4d315661be183e86132 of the engine or a more recent one (note: stacktrace below was created from the most recent development version, 3f326698831ebdb4f6772113f55b5c5f1c067e73 at the time)
2. Get the o3de-atomtest project (which contains a DiffuseProbeGrid)
3. Build the Editor with the 'profile' config
4. Let the Asset Processor rebuild the assets of the project
5. Run the Editor or Material Editor in the o3de-atomtest project with the Vulkan RHI, which triggers the segfault.

**Backtrace**
This is the backtrace from gdb when starting the Material Editor from commit 3f326698831ebdb4f6772113f55b5c5f1c067e73

```
#0  0x0000000000000000 in ?? ()
#1  0x00007fffda1e2a0f in AZ::Vulkan::RayTracingBlas::CreateBuffersInternal (this=<optimized out>, deviceBase=..., descriptor=0x7fffffff8940, bufferPools=...)
    at /data/roald/o3de/Gems/Atom/RHI/Vulkan/Code/Source/RHI/RayTracingBlas.cpp:104
#2  0x00007fffd8410162 in AZ::RHI::RayTracingBlas::CreateBuffers (this=0x7ffefc72f660, device=..., descriptor=0x7ffefc72f798, rayTracingBufferPools=...)
    at /data/roald/o3de/Gems/Atom/RHI/Code/Source/RHI/RayTracingAccelerationStructure.cpp:122
#3  0x00007fffd83b59cd in AZ::Render::DiffuseProbeGridFeatureProcessor::OnVisualizationModelAssetReady (this=this@entry=0x7ffefd3b8390, asset=...)
    at /data/roald/o3de/Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGridFeatureProcessor.cpp:733
#4  0x00007fffd83b6000 in AZ::Render::DiffuseProbeGridFeatureProcessor::OnAssetReady (this=0x7ffefd3b8390, asset=...)
    at /data/roald/o3de/Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGridFeatureProcessor.cpp:758
#5  0x00007fffd83b6157 in non-virtual thunk to AZ::Render::DiffuseProbeGridFeatureProcessor::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData>) ()
    at /data/roald/o3de/Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/BufferScopeAttachmentDescriptor.h:20
#6  0x0000555558419de2 in AZStd::Internal::INVOKE<void (AZ::Data::AssetEvents::*&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Internal::HandlerNode<AZ::Data::AssetEvents, AZ::Data::AssetEvents, AZ::Internal::EBusContainer<AZ::Data::AssetEvents, AZ::Data::AssetEvents, (AZ::EBusAddressPolicy)1, (AZ::EBusHandlerPolicy)1>::HandlerHolder, true>&, AZ::Data::Asset<AZ::Data::AssetData>&, void> (f=<optimized out>, arg0=..., args=...)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/typetraits/invoke_traits.h:177
#7  0x00005555583e4445 in AZStd::invoke<void (AZ::Data::AssetEvents::*&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Internal::HandlerNode<AZ::Data::AssetEvents, AZ::Data::AssetEvents, AZ::Internal::EBusContainer<AZ::Data::AssetEvents, AZ::Data::AssetEvents, (AZ::EBusAddressPolicy)1, (AZ::EBusHandlerPolicy)1>::HandlerHolder, true>&, AZ::Data::Asset<AZ::Data::AssetData>&> (
    f=@0x7fffffffc1c0: &virtual AZ::Data::AssetEvents::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData>), args=..., args=...) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/invoke.h:39
#8  AZ::EBusEventProcessingPolicy::Call<void (AZ::Data::AssetEvents::*&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Internal::HandlerNode<AZ::Data::AssetEvents, AZ::Data::AssetEvents, AZ::Internal::EBusContainer<AZ::Data::AssetEvents, AZ::Data::AssetEvents, (AZ::EBusAddressPolicy)1, (AZ::EBusHandlerPolicy)1>::HandlerHolder, true>&, AZ::Data::Asset<AZ::Data::AssetData>&> (
    func=@0x7fffffffc1c0: &virtual AZ::Data::AssetEvents::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData>), iface=..., args=...) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/EBus/Policies.h:437
#9  AZ::Internal::EBusContainer<AZ::Data::AssetEvents, AZ::Data::AssetEvents, (AZ::EBusAddressPolicy)1, (AZ::EBusHandlerPolicy)1>::Dispatcher<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::Event<void (AZ::Data::AssetEvents::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::Asset<AZ::Data::AssetData>&> (id=..., func=@0x7fffffffc1c0: &virtual AZ::Data::AssetEvents::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData>), args=...)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/EBus/Internal/BusContainer.h:149
#10 0x00005555583d5024 in AZ::Data::AssetManager::NotifyAssetReady (this=<optimized out>, asset=...) at /data/roald/o3de/Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:1703
#11 0x000055555841b381 in AZStd::Internal::INVOKE<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData>, void> (f=<optimized out>, arg0=<optimized out>, 
    args=...) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/typetraits/invoke_traits.h:177
#12 0x000055555841b27e in AZStd::invoke<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData> > (f=
    @0x55555a2d8320: (void (AZ::Data::AssetManager::*)(AZ::Data::AssetManager * const, AZ::Data::Asset<AZ::Data::AssetData>)) 0x55555a0bddd0, this adjustment 140733193388035, args=..., args=...)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/invoke.h:39
#13 AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::QueueFunction<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData> >(void (AZ::Data::AssetManager::*&&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*&&, AZ::Data::Asset<AZ::Data::AssetData>&&)::{lambda()#1}::operator()() (this=0x55555a2d8320) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/EBus/BusImpl.h:838
#14 AZStd::Internal::INVOKE<AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::QueueFunction<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData> >(void (AZ::Data::AssetManager::*&&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*&&, AZ::Data::Asset<AZ::Data::AssetData>&&)::{lambda()#1}&> (f=...) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/typetraits/invoke_traits.h:208
#15 AZStd::invoke<AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::QueueFunction<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData> >(void (AZ::Data::AssetManager::*&&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*&&, AZ::Data::Asset<AZ::Data::AssetData>&&)::{lambda()#1}&>(void (AZ::Data::AssetManager::*&&)(AZ::Data::Asset<AZ::Data::AssetData>)) (f=...) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/invoke.h:39
#16 AZStd::Internal::function_util::invoke_void_return_wrapper<void>::call<AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::QueueFunction<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData> >(void (AZ::Data::AssetManager::*&&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*&&, AZ::Data::Asset<AZ::Data::AssetData>&&)::{lambda()#1}&>(AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::QueueFunction<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData> >(void (AZ::Data::AssetManager::*&&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*&&, AZ::Data::Asset<AZ::Data::AssetData>&&)::{lambda()#1}&) (args=...) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/function_template.h:40
#17 AZStd::Internal::function_util::get_invoker<void (), AZStd::Internal::function_util::function_obj_tag, AZ::Internal::EBusEnvironmentAllocator>::call<AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::QueueFunction<void (AZ::Data::AssetManager::*)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*, AZ::Data::Asset<AZ::Data::AssetData> >(void (AZ::Data::AssetManager::*&&)(AZ::Data::Asset<AZ::Data::AssetData>), AZ::Data::AssetManager*&&, AZ::Data::Asset<AZ::Data::AssetData>&&)::{lambda()#1}>(AZStd::Internal::function_util::function_buffer&) (function_obj_ptr=...)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/function_template.h:146
#18 0x0000555558412061 in AZStd::function_intermediate<void>::operator() (this=<optimized out>) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/function_template.h:603
#19 AZStd::function<void ()>::operator()() const (this=<optimized out>) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/function_template.h:683
#20 AZ::EBusQueuePolicy<true, AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZStd::recursive_mutex>::Execute (this=<optimized out>, this@entry=0x555559a9fe08)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/EBus/Policies.h:266
#21 0x00005555583d1a4a in AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::Data::AssetEvents, AZ::Data::AssetEvents>, AZ::BusInternal::EBusImplTraits<AZ::Data::AssetEvents, AZ::Data::AssetEvents> >::ExecuteQueuedEvents ()
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/EBus/BusImpl.h:344
#22 AZ::Data::AssetManager::DispatchEvents (this=<optimized out>) at /data/roald/o3de/Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:573
#23 0x00005555583ee092 in AZ::Data::WaitForAsset::Wait (this=this@entry=0x7fffffffc420) at /data/roald/o3de/Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:324
#24 0x00005555583e35c3 in AZ::Data::WaitForAsset::WaitUntilReady (this=this@entry=0x7fffffffc420) at /data/roald/o3de/Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:304
#25 0x00005555583c39f4 in AZ::Data::AssetManager::BlockUntilLoadComplete (this=0x7fff7bd8f7d0, asset=...) at /data/roald/o3de/Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:831
#26 0x00005555583c3956 in AZ::Data::AssetInternal::BlockUntilLoadComplete (asset=...) at /data/roald/o3de/Code/Framework/AzCore/AzCore/Asset/AssetCommon.cpp:110
#27 0x0000555555b04334 in AZ::Data::Asset<AZ::RPI::MaterialAsset>::BlockUntilLoadComplete (this=<optimized out>, this@entry=0x7fffffffc660) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/Asset/AssetCommon.h:1144
#28 0x0000555555a2a08c in AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::MaterialAsset> (productPath=0x55555847789b "materials/special/shadowcatcher.azmaterial", reporting=reporting@entry=AZ::RPI::AssetUtils::TraceLevel::Error)
    at /data/roald/o3de/Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Asset/AssetUtils.h:94
#29 0x0000555555a1682a in MaterialEditor::MaterialViewportWidget::MaterialViewportWidget (this=this@entry=0x55555a1f0010, parent=<optimized out>)
    at /data/roald/o3de/Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp:193
#30 0x0000555555a19a8a in MaterialEditor::MaterialEditorWindow::MaterialEditorWindow (this=0x7ffefcf0a430, parent=<optimized out>) at /data/roald/o3de/Gems/Atom/Tools/MaterialEditor/Code/Source/Window/MaterialEditorWindow.cpp:73
#31 0x0000555555b3e8f5 in MaterialEditor::MaterialEditorApplication::CreateMainWindow (this=0x7fffffffcf90) at /data/roald/o3de/Gems/Atom/Tools/MaterialEditor/Code/Source/MaterialEditorApplication.cpp:125
#32 0x0000555555b3ebfa in non-virtual thunk to MaterialEditor::MaterialEditorApplication::CreateMainWindow() () at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/any.h:273
#33 0x0000555555bed7f5 in AZStd::Internal::INVOKE<void (AtomToolsFramework::AtomToolsMainWindowFactoryRequests::*)(), AtomToolsFramework::AtomToolsMainWindowFactoryRequests*&, , void> (f=@0x7fffffffcda0: &virtual table offset 0, 
    arg0=<optimized out>) at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/typetraits/invoke_traits.h:177
#34 AZStd::invoke<void (AtomToolsFramework::AtomToolsMainWindowFactoryRequests::*)(), AtomToolsFramework::AtomToolsMainWindowFactoryRequests*&> (f=@0x7fffffffcda0: &virtual table offset 0, args=<optimized out>)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/std/function/invoke.h:39
#35 AZ::EBusEventProcessingPolicy::Call<void (AtomToolsFramework::AtomToolsMainWindowFactoryRequests::*)(), AtomToolsFramework::AtomToolsMainWindowFactoryRequests*&> (func=@0x7fffffffcda0: &virtual table offset 0, iface=<optimized out>)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/EBus/Policies.h:437
#36 AZ::Internal::EBusContainer<AtomToolsFramework::AtomToolsMainWindowFactoryRequests, AtomToolsFramework::AtomToolsMainWindowFactoryRequests, (AZ::EBusAddressPolicy)0, (AZ::EBusHandlerPolicy)0>::Dispatcher<AZ::EBus<AtomToolsFramework::AtomToolsMainWindowFactoryRequests, AtomToolsFramework::AtomToolsMainWindowFactoryRequests> >::Broadcast<void (AtomToolsFramework::AtomToolsMainWindowFactoryRequests::*)()> (func=@0x7fffffffcda0: &virtual table offset 0)
    at /data/roald/o3de/Code/Framework/AzCore/./AzCore/EBus/Internal/BusContainer.h:1540
#37 0x0000555555bdb89c in AtomToolsFramework::AtomToolsApplication::StartCommon (this=0x7fffffffcf90, systemEntity=<optimized out>) at /data/roald/o3de/Gems/Atom/Tools/AtomToolsFramework/Code/Source/Application/AtomToolsApplication.cpp:204
#38 0x0000555555b3e2a1 in MaterialEditor::MaterialEditorApplication::StartCommon (this=0x55555a2d8320, systemEntity=0x1) at /data/roald/o3de/Gems/Atom/Tools/MaterialEditor/Code/Source/MaterialEditorApplication.cpp:102
#39 0x0000555557acb08c in AzFramework::Application::Start (this=0x7fffffffcf90, descriptor=..., startupParameters=...) at /data/roald/o3de/Code/Framework/AzFramework/AzFramework/Application/Application.cpp:196
#40 0x0000555555b14784 in main (argc=1, argv=0x7fffffffe358) at /data/roald/o3de/Gems/Atom/Tools/MaterialEditor/Code/Source/main.cpp:18
```

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Linux (Ubuntu 20.04 LTS)
 - CPU: i7-11850H
 - GPU: NVidia RTX 3080 laptop [EDIT: At the time of the stacktrace this GPU was probably nonfunctional due to a sneaky driver upgrade -- the segfault no longer triggers on this hardware, but it does still trigger on an older machine with a GTX 980]
 - Memory: 16GB

**Comment by RoaldFre** [comment]
Addendum: it may have triggered as a result of a sneaky driver update, as I cannot seem to reproduce it after a reboot on that hardware (RTX 3080). But it does still trigger on another machine with a GTX 980.

**Comment by jeremyong-az** [comment]
I believe this should be fixed now that #7544 is merged, feel free to reopen if this is not the case, thanks!
